frontend/src/core/panels/Panels.tsx

```diff
@@ -10,70 +10,216 @@
  * Contributors:
  *     Obeo - initial API and implementation
  *******************************************************************************/
-import React, { useState } from 'react';
-import styles from './Panels.module.css';
+import { makeStyles } from '@material-ui/core/styles';
+import React, { useRef, useState } from 'react';
+import { Site } from 'workbench/Site';
 import { PanelsProps } from './Panels.types';
 
+const MIN_PANEL_WIDTH: number = 42;
+const MAIN_AREA_MIN_WIDTH: number = 100;
+const RESIZER_WIDTH: number = 4;
+
+const usePanelStyles = makeStyles((theme) => ({
+  panel: {
+    display: 'grid',
+    gridTemplateTows: 'minmax(0, 1fr)',
+    gridTemplateColumns: 'minmax(0, 1fr)',
+  },
+  verticalResizer: {
+    display: 'grid',
+    width: `${RESIZER_WIDTH}px`,
+    cursor: 'col-resize',
+    '& div': {
+      backgroundColor: theme.palette.divider,
+      borderColor: theme.palette.divider,
+      borderRightStyle: 'solid',
+      borderRightWidth: '1px',
+    },
+  },
+}));
+
+interface PanelState {
+  isDragging: boolean;
+  initialPosition: number;
+  expanded: boolean;
+  resizablePanelSize: number;
+}
+
 export const Panels = ({
-  firstPanel,
-  secondPanel,
-  resizablePanel = 'FIRST_PANEL',
-  initialResizablePanelSize,
+  editingContextId,
+  selection,
+  setSelection,
+  readOnly,
+  leftContributions,
+  rightContributions,
+  mainArea,
+  leftPanelInitialSize,
+  rightPanelInitialSize,
 }: PanelsProps) => {
-  const initialState = { isDragging: false, initialPosition: 0, resizablePanelSize: initialResizablePanelSize };
-  const [state, setState] = useState(initialState);
-  const { isDragging, resizablePanelSize } = state;
+  const ref = useRef<HTMLDivElement>(null);
+  const [leftPanelState, setLeftPanelState] = useState<PanelState>({
+    isDragging: false,
+    initialPosition: 0,
+    expanded: true,
+    resizablePanelSize: leftPanelInitialSize,
+  });
+  const [rightPanelState, setRightPanelState] = useState<PanelState>({
+    isDragging: false,
+    initialPosition: 0,
+    expanded: true,
+    resizablePanelSize: rightPanelInitialSize,
+  });
 
-  const startResize = (event) => {
-    let initialPosition = event.clientX;
-    setState((prevState) => {
-      return { isDragging: true, initialPosition, resizablePanelSize: prevState.resizablePanelSize };
+  const startResizeLeft: React.MouseEventHandler = (event) => {
```

Comment:
`React.MouseEventHandler<HTMLDivElement>` (this way the event will have the proper type for its target)